src/Features/CSharp/Portable/Completion/CSharpCompletionService.cs

```diff
@@ -78,17 +78,24 @@ public override CompletionRules GetRules()
         {
             var options = _workspace.Options;
 
-            var rule = options.GetOption(CompletionOptions.EnterKeyBehavior, LanguageNames.CSharp);
+            var enterRule = options.GetOption(CompletionOptions.EnterKeyBehavior, LanguageNames.CSharp);
+            var snippetRule = options.GetOption(CompletionOptions.SnippetsBehavior, LanguageNames.CSharp);
 
             // Although EnterKeyBehavior is a per-language setting, the meaning of an unset setting (Default) differs between C# and VB
             // In C# the default means Never to maintain previous behavior
-            if (rule == EnterKeyRule.Default)
+            if (enterRule == EnterKeyRule.Default)
             {
-                rule = EnterKeyRule.Never;
+                enterRule = EnterKeyRule.Never;
+            }
+
+            if (snippetRule == SnippetsRule.Default)
+            {
+                snippetRule = SnippetsRule.AlwaysInclude;
             }
 
             // use interlocked + stored rules to reduce # of times this gets created when option is different than default
-            var newRules = _latestRules.WithDefaultEnterKeyRule(rule);
+            var newRules = _latestRules.WithDefaultEnterKeyRule(enterRule)
+                                       .WithSnippetsRule(snippetRule);
 
             Interlocked.Exchange(ref _latestRules, newRules);
```

Comment:
It's not really clear that this is saving any allocations, but I guess it is because most of the time `WithDefaultEnterKeyRule` and `WithSnippetsRule` return the same instance that they are called on and a new instance is only allocated when they actually change something?